stats/collector.go

```diff
@@ -201,23 +194,31 @@ func makeMetrics(counters keyIndex) []Metric {
 	return metrics
 }
 
-func (s *BufferedCollector) CollectMetadata(accountMetadata map[string]string) {
-	entries := make([]MetadataEntry, len(accountMetadata))
-	i := 0
-	for k, v := range accountMetadata {
-		entries[i] = MetadataEntry{Name: k, Value: v}
-		i++
-	}
+func (s *BufferedCollector) CollectMetadata(accountMetadata *Metadata) {
 	ctx, cancel := context.WithTimeout(context.Background(), s.sendTimeout)
 	defer cancel()
-	err := s.sender.UpdateMetadata(ctx, Metadata{
-		InstallationID: s.getInstallationID(),
-		Entries:        entries,
-	})
+	err := s.sender.UpdateMetadata(ctx, *accountMetadata)
```

Comment:
couldn't this lead to a null dereference if the account metadata is null? this can happen if reading the account id fails no?

Reply:
no, if reading the account id fails, the `cloudMetadataProvider.GetMetadata()` returns nil, and then it will be ignored when creating the unified metadata object (serach for cloudMetadataProvider.GetMetadata())